### ISSUE TITLE
Date Regex

### ISSUE THREAD

**Issue by david-buderus** [upb-uc4/lagom-core#342]
**Is your feature request related to a problem? Please describe.**
Our currently date check is a non-readable and not  maintainable regex.
```scala
val dateRegex: Regex = """^(?:(?:(?:(?:(?:[1-9]\d)(?:0[48]|[2468][048]|[13579][26])|(?:(?:[2468][048]|[13579][26])00))(-)(?:0?2\1(?:29)))|(?:(?:[1-9]\d{3})(-)(?:(?:(?:0?[13578]|1[02])\2(?:31))|(?:(?:0?[13-9]|1[0-2])\2(?:29|30))|(?:(?:0?[1-9])|(?:1[0-2]))\2(?:0?[1-9]|1\d|2[0-8])))))$""".r
```

**Describe the solution you'd like**
We could use SimpleDateFormat, and expect that java does this check right.
```scala
def validate(date: String) = try {
    val df = new SimpleDateFormat("yyyy-MM-dd")
    df.setLenient(false)
    df.parse(date)
    true
  } catch {
    case e: ParseException => false
  }
```

**Describe alternatives you've considered**
We could use JodaTime (Threadsafe)
```scala
import org.joda.time.format._
val fmt = DateTimeFormat forPattern "yyyy-MM-dd"

def parseDate(input: String) = try {
  Some(fmt parseDateTime input)
} catch {
  case e: IllegalArgumentException => None
}
```

**Comment by lrschaef** [comment]
Since a few weeks I wanted to work on this regex to make it readable. If I find the time I want to try this before we completely switch the validation method. Regex checks are very useful, because our frontend wants to know the validation regexes (obviously it doesnt help if the regex is completely unreadable, I agree on that).

**Comment by lrschaef** [comment]
The more readable version of the regex is implemented in #346
